### PR TITLE
Docs updates for 0.6.0

### PR DESCRIPTION
**Pull request by bgilbert**
Add migration notes for spec 1.1.0, spec bump instructions, and update NEWS.

### CODE REVIEW

File: docs/migrating-configs.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
+# Migrating Between Configuration Versions
+
+Occasionally, there are changes made to Fedora CoreOS configuration that break backward compatibility. While this is not a concern for running machines (since Ignition only runs one time during first boot), it is a concern for those who maintain configuration files. This document serves to detail each of the breaking changes and tries to provide some reasoning for the change. This does not cover all of the changes to the spec - just those that need to be considered when migrating from one version to the next.
+
+## From Version 1.0.0 to 1.1.0
+
+There are not any breaking changes between versions 1.0.0 and 1.1.0 of the configuration specification. Any valid 1.0.0 configuration can be updated to a 1.1.0 configuration by simply changing the version string in the config.
```
Comment (ashcrow) [inline docs/migrating-configs.md:7]: nit: -e `s|not any|no|`-e `s|simply||`
```suggestion
There are no breaking changes between versions 1.0.0 and 1.1.0 of the configuration specification. Any valid 1.0.0 configuration can be updated to a 1.1.0 configuration by changing the version string in the config.
```

File: docs/migrating-configs.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+version: 1.1.0
+storage:
+  files:
+    - path: /etc/hosts
+      mode: 0644
+      contents:
+        source: https://example.com/etc/hosts
+        http_headers:
+          - name: Authorization
+            value: Basic YWxhZGRpbjpvcGVuc2VzYW1l
```
Comment (ashcrow) [inline docs/migrating-configs.md:180]: Ha! Traditional example :stuck_out_tongue:

**Comment by bgilbert** [comment]
Updated, and @zonggen has given his okay.  Merging.

File: NEWS [hunk 3]
```diff
@@ -1,3 +1,31 @@
...
+    - Add inline field to TLS certificate authorities and config merge and
+      replace (1.1.0)
+    - Add local field for embedding contents from local file (1.1.0)
+    - Add storage.trees section for embedding local directory trees (1.1.0)
+    - Auto-select smallest encoding for inline or local contents (1.1.0)
+    - Add http_headers field for specifying HTTP headers on fetch (1.1.0)
+
+  Bug Fixes:
+
+    - Include mount options in generated mount units (1.1.0)
```
Comment (arithx) [inline NEWS:17]: Why is this a bug fix instead of a feature?

Comment (bgilbert) [reply NEWS:17]: As discussed OOB, the `mount_options` and `with_mount_unit` fields both already existed, but the options were being omitted from the generated unit.

**Pull request merged**